experimental/dds/tree2/src/shared-tree-core/editManager.ts

```diff
@@ -482,6 +582,18 @@ export class EditManager<
 				true,
 			);
 			this.localBranch.rebaseOnto(this.trunk);
+
+			if (this.oldestRevertibleSequenceId === undefined) {
```

Comment:
Does it work to simplify this section to:

```ts
this.oldestRevertibleSequenceId = this.getOldestRevertibleSequenceId(this.localBranch);
```

And just do the event registration in the constructor?

Reply:
This part is to reduce unnecessary computation since we know we don't have to update this value unless the oldest revertible gets disposed. I'll move the event registration though.

Reply:
Oh I see. Good thinking!

If you want to reduce unnecessary computation maximally (and have less code), I have a suggestion: try the "lazy cache" pattern. Sort of like a "getOrCreate". If `this.oldestRevertibleSequenceId` is already cached, then `this.getOldestRevertibleSequenceId()` returns the cached value. If it's not cached, then the getter computes it, caches it, and returns it.

* In trunk eviction, you call `this.getOldestRevertibleSequenceId()` and only take action if it's defined.
* On discard of the oldest revertible, you simply clear the cache (`this.oldestRevertibleSequenceId = undefined`).
* You don't need to do anything here in `addSequencedChange`.

As a bonus this makes the computation lazy. If we had a flag that turned off trunk eviction, for example, we wouldn't ever do any of this work since it'd only be done on demand in `this.trimTrunk`.

Reply:
ooo good idea